app/electron/plugin-management.ts

```diff
@@ -446,20 +457,111 @@ async function downloadExtractArchive(
     throw new Error('Download cancelled');
   }
 
+  // Create temporary folder for extraction
   const tempDir = await fs.mkdtempSync(path.join(os.tmpdir(), 'headlamp-plugin-temp-'));
   // Defaulting to '' should never happen if recursive is true. So this is for the type
   // checker only.
   const tempFolder = fs.mkdirSync(path.join(tempDir, pluginName), { recursive: true }) ?? '';
 
+  // First, download and extract the main archive
   if (progressCallback) {
-    progressCallback({ type: 'info', message: 'Downloading Plugin' });
+    progressCallback({ type: 'info', message: 'Downloading main plugin archive' });
   }
+
+  await downloadAndExtractSingleArchive(
+    pluginInfo.archiveURL,
+    pluginInfo.archiveChecksum,
+    tempFolder,
+    progressCallback,
+    signal
+  );
+
+  // Check if there's also a platform-specific archive
+  const platform = os.platform();
+  const arch = os.arch();
+  const platformArchKey = getPlatformArchKey(platform, arch);
+  const platformArchChecksumKey = getPlatformArchChecksumKey(platform, arch);
+
+  // If platform-specific archive exists, download and extract it too
+  if (pluginInfo[platformArchKey]) {
+    if (progressCallback) {
+      progressCallback({
+        type: 'info',
+        message: `Downloading platform-specific archive for ${platform}:${arch}`,
+      });
+    }
+
+    await downloadAndExtractSingleArchive(
+      pluginInfo[platformArchKey],
+      pluginInfo[platformArchChecksumKey],
```

Comment:
what if the users arch isn't supported by the plugin? We have that before accessing the key